xarray/backends/scipy_.py

```diff
@@ -37,30 +37,39 @@ def __init__(self, variable_name, datastore):
 
     @property
     def array(self):
-        return self.datastore.ds.variables[self.variable_name].data
+        with self.datastore.ensure_open(autoclose=True):
+            return self.datastore.ds.variables[self.variable_name].data
 
     @property
     def dtype(self):
-        # always use native endianness
-        return np.dtype(self.array.dtype.kind + str(self.array.dtype.itemsize))
+        with self.datastore.ensure_open(autoclose=True):
```

Comment:
We need to use `NumpyIndexAdapater` for scipy (which uses numpy arrays directly in it's netcdf reader), but not for netCDF4 (which uses it's own array class with different "orthogonal" indexing behavior)